### PR TITLE
#77383 - include http verb for route and replace

### PR DESCRIPTION
**Pull request by mszymeshop**
### What
Route and Replace incorrectly used HTTP verb for its requests. The current PR changes that by using the HTTP verb defined for the particular endpoint.

### CODE REVIEW

File: src/CaptainHook.EventHandlerActor/Handlers/Requests/DefaultRequestBuilder.cs [hunk 1]
```diff
@@ -179,7 +179,7 @@ public string BuildPayload(WebhookConfig config, string sourcePayload, IDictiona
 
         /// <inheritdoc />
 
-        public HttpMethod SelectHttpMethod(WebhookConfig webhookConfig, string payload)
+        public virtual HttpMethod SelectHttpMethod(WebhookConfig webhookConfig, string payload)
```
Comment (chomick) [inline src/CaptainHook.EventHandlerActor/Handlers/Requests/DefaultRequestBuilder.cs:182]: Have you checked for all other usages of RuleAction.Route and if it should consider the new the as well?

File: src/CaptainHook.EventHandlerActor/Handlers/Requests/RouteAndReplaceRequestBuilder.cs [hunk 2]
```diff
@@ -55,6 +56,11 @@ protected override WebhookConfig SelectWebhookConfigCore(WebhookConfig webhookCo
             return routeConfig;
         }
 
+        public override HttpMethod SelectHttpMethod(WebhookConfig webhookConfig, string payload) =>
+            SelectWebhookConfig(webhookConfig, payload).HttpMethod;
```
Comment (chomick) [inline src/CaptainHook.EventHandlerActor/Handlers/Requests/RouteAndReplaceRequestBuilder.cs:60]: Should we not simply replace the implementation of the SelectHttpMethod in base class to use virtual SelectWebhookConfig?

Comment (mszymeshop) [reply src/CaptainHook.EventHandlerActor/Handlers/Requests/RouteAndReplaceRequestBuilder.cs:60]: Unfortunately, they differ slightly + there is code duplication in DefaultRequestBuilder which requires additional checks and tests. Nothing major, but the ultimate goal was not to touch DefaultRequestBuilder too much for its core role in the product. Something we can definitely improve and refactor in the future but I didn't feel I have time to take care of this at the moment.

File: src/CaptainHook.EventHandlerActor/Handlers/Requests/DefaultRequestBuilder.cs (hunk 1 above)
Comment (mszymeshop) [reply src/CaptainHook.EventHandlerActor/Handlers/Requests/DefaultRequestBuilder.cs:182]: As far as IRequestBuilder is concerned, I believe I have yes. 

> if it should consider the new the as well
Could you say more about this?

**Pull request merged**